staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -356,6 +373,13 @@ func (o *ApplyOptions) Run() error {
 		return fmt.Errorf("no objects passed to apply")
 	}
 
+	if o.preprocessFn != nil {
```

Comment:
We would want the `objects` field to be populated (from the resource builder) before the pre-processor, so that the pre-processor could work on the `objects`.